File: November/Day9/Online_Stock_Span.java
class StockSpanner {
    Stack<Pair<Integer,Integer>> st = new Stack<>();

    public StockSpanner() {
        
    }
    
    public int next(int price) {
        if(st.isEmpty()){
            st.push(new Pair(price,1));
            return 1;
        }
        int ans=1;
        while(!st.isEmpty() && st.peek().getKey()<=price){
            Pair<Integer,Integer> p=st.pop();
            ans+=p.getValue();
        }
        st.push(new Pair(price,ans));
        return ans;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: November/Day9/Online_Stock_Span.cpp
class StockSpanner {
public:
    stack<pair<int,int>>st;
    StockSpanner() {
        
    }
    
    int next(int price) {
     if(st.empty())
     {
         st.push({price,1});
         return 1;
     }
     else
     {
         int ans = 1;
         while(!st.empty() && st.top().first<=price)
         {
             auto x = st.top();
             st.pop();
             int first = x.first;
             int second = x.second;
             ans+=second;
         }
         st.push({price,ans});
         return ans;
     }
        return 0;
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */